### PR TITLE
- changing mermaid suport

### PR DESCRIPTION
**Pull request by franzhaas**
the already existing mermaid support, did not work for me.

It showed me "syntax error bomb displays", which where displayed just milliseconds after the correctly rendered graph was shown.

I added the reveal mermaid plugin, and removed the already existing mermaid initialisation.

While this does work for me, it is probably quite a change and i really would like to get some input on how to continue, or change my environment to fit the preexisting solution (which is still used for the hard parts, i basically swap pure mermaid for the plugin, as a result of trial and error)

### CODE REVIEW

File: share/templates/reveal/index.html.j2 [hunk 1]
```diff
@@ -145,21 +141,21 @@ require(
     },
     [
       "{{ reveal_url_prefix }}/dist/reveal.js",
-      "{{ reveal_url_prefix }}/plugin/notes/notes.js"
+      "{{ reveal_url_prefix }}/plugin/notes/notes.js",
+      "https://cdn.jsdelivr.net/npm/reveal.js-mermaid-plugin@2.2.0/plugin/mermaid/mermaid.js"
```
Comment (bollwyvl) [inline share/templates/reveal/index.html.j2:145]: I would recommend hoisting this to a configurable on the python side:

```py
# nbconvert/exporters/slides.py
reveal_url_prefix = Unicode(
  help="..."
).tag(config=True)

reveal_mermaid_js_url = Unicode(
  default="https://cdn.jsdelivr.net/npm/reveal.js-mermaid-plugin@2.2.0/plugin/mermaid/mermaid.js",
  help="...",
).tag(config=True)
```

```js
[
      "{{ reveal_url_prefix }}/dist/reveal.js",
      "{{ reveal_url_prefix }}/plugin/notes/notes.js",
      {% if reveal_mermaid_js_url %}"{{ reveal_mermaid_js_url }}",{% endif %}
]
```

And then below:

```js
plugins: [
  RevealNotes,
  {% if reveal_mermaid_js_url %}RevealMermaid,{% endif %}
]
```

Comment (franzhaas) [reply share/templates/reveal/index.html.j2]: ok thanks, makes sense, I did it.

Any further things I can do or should know?

Thanks again,
Franz

File: share/templates/reveal/index.html.j2 [hunk 2]
```diff
@@ -145,21 +141,24 @@ require(
     },
     [
       "{{ reveal_url_prefix }}/dist/reveal.js",
-      "{{ reveal_url_prefix }}/plugin/notes/notes.js"
+      "{{ reveal_url_prefix }}/plugin/notes/notes.js",
+      {% if reveal_mermaid_js_url %}"{{ reveal_mermaid_js_url }}",{% endif %},
     ],
 
-    function(Reveal, RevealNotes){
+    function(Reveal, RevealNotes, RevealMermaid){
```
Comment (bollwyvl) [inline share/templates/reveal/index.html.j2:148]: sorry, missed this: the positional argument would need to be conditional as well:

```js
function(
    Reveal, 
    RevealNotes, 
    {% if reveal_mermaid_js_url %}RevealMermaid,{% endif %}
){
```

Also feel free to normalize the whitespace: the "floating indent" style isn't necessary, just consistency with what's around (in this case four spaces).

Comment (franzhaas) [reply share/templates/reveal/index.html.j2]: good idea, i also did that...

Comment (franzhaas) [reply share/templates/reveal/index.html.j2]: Can someone please assign the improvement label? I am sorry, i either don't have the right to set it, or i just don't find the feature...

File: share/templates/reveal/index.html.j2 [hunk 3]
```diff
@@ -145,21 +143,26 @@ require(
     },
     [
       "{{ reveal_url_prefix }}/dist/reveal.js",
-      "{{ reveal_url_prefix }}/plugin/notes/notes.js"
+      "{{ reveal_url_prefix }}/plugin/notes/notes.js",
+      {% if reveal_mermaid_js_url %}"{{ reveal_mermaid_js_url }}",{% endif %},
```
Comment (bollwyvl) [inline share/templates/reveal/index.html.j2:147]: there's an extra comma after the `endif`

File: nbconvert/exporters/slides.py [hunk 4]
```diff
@@ -119,6 +119,12 @@ def _template_extension_default(self):
         """
     ).tag(config=True)
 
+    reveal_mermaid_js_url = Unicode(
+        help="specifies the mermaid reveal plugin url. defaults to https://unpkg.com/reveal.js-mermaid-plugin@2.2.0",
```
[the remark this answers is not in the document]
Comment (franzhaas) [reply nbconvert/exporters/slides.py]: I am unsure if i understood you correctly... do you mean those @default()?

anyhow i tried..

**Comment by franzhaas** [comment]
Is there still something I need to do? Something i overlooked?

Thanks,
Franz

**Comment by franzhaas** [comment]
- i believe to much changes happened in the meantime...

**Pull request closed without merging**